lib/fog/azurerm/models/storage/blobs.rb

```diff
@@ -9,6 +9,23 @@ class Blobs < Fog::Collection
         model Fog::Storage::AzureRM::Blob
         attribute :container_name
 
+        def all(options = { metadata: true })
+          blobs = []
+          service.list_blobs(container_name, options).each do |blob|
+            hash = Blob.parse blob
+            hash['container_name'] = container_name
+            blobs << hash
+          end
+          load blobs
+        end
+
+        def get(identity)
```

Comment:
'get' and 'all' method should be independent. We suggest to implement separate request to get blob. 
We are updating this in other places in code base as well.